### ISSUE TITLE
Add VkPhysicalDevicePortabilitySubsetFeaturesKHR:: shaderIntCubeSampler capability

### ISSUE THREAD

**Issue by billhollings** [KhronosGroup/Vulkan-Portability#43]
Some CTS tests fail when the shader samples integer cube textures. 

This might be fixable with effort within the implementations.

In the meantime, these use cases can be disabled in CTS and reported as Not Supported, by adding a `VkPhysicalDevicePortabilitySubsetFeaturesKHR:: shaderIntCubeSampler` capability.

This affects 8 CTS tests:

```
dEQP-VK.glsl.texture_functions.texturegrad.samplercube_fixed_vertex
dEQP-VK.glsl.texture_functions.texturegrad.samplercube_fixed_fragment
dEQP-VK.glsl.texture_functions.texturegrad.isamplercube_vertex
dEQP-VK.glsl.texture_functions.texturegrad.isamplercube_fragment
dEQP-VK.glsl.texture_functions.texturegrad.samplercubearray_fixed_vertex
dEQP-VK.glsl.texture_functions.texturegrad.samplercubearray_fixed_fragment
dEQP-VK.glsl.texture_functions.texturegrad.isamplercubearray_vertex
dEQP-VK.glsl.texture_functions.texturegrad.isamplercubearray_fragment
  Fail (Image mismatch)
```

**Comment by cdavis5e** [comment]
So it works with float and *unsigned* integer textures, but it breaks with normalized and *signed* integer textures... This seems to me to be a bug in Metal--or at least, in the Metal driver for Apple Silicon. In fact... I seem to recall filing a feedback for this a while back.

**Comment by billhollings** [comment]
> This seems to me to be a bug in Metal

If so, then we can add this to the exceptions list, instead of adding a flag.

**Comment by cdavis5e** [comment]
Looking at this a bit further, I noticed something else: all the failing tests set only ds/dx or ds/dy, but the passing tests set dt/dx or dt/dy (and not ds/dx or ds/dy). I wonder what the cube array tests show...

(This is, of course, purely for the purposes of filing another feedback; since I filed the original feedback back when I was with CW, I no longer have access to it.)

**Comment by cdavis5e** [comment]
I have confirmed that it has nothing at all to do with the pixel format and everything to do with which derivatives are set in the gradient.

**Comment by cdavis5e** [comment]
Based on @lexaknyazev's code, I've added a workaround for this problem.

**Comment by theFJ1991** [comment]
F J 9 1On 3 Dec 2023, at 08:46, Chip Davis ***@***.***> wrote:﻿
Closed #43 as completed.
